comment/models/comments.py

```diff
@@ -58,8 +57,25 @@ def dislikes(self):
     @property
     def is_flagged(self):
         if hasattr(self, 'flag'):
-            self.flag.refresh_from_db()
-            allowed_flags = getattr(settings, 'COMMENT_FLAGS_ALLOWED', 0)
-            if allowed_flags:
-                return self.flag.count > allowed_flags
+            if not self.flag.is_flag_enabled:
+                return False
+            return self.flag.state != self.flag.UNFLAGGED
+        return False
```

Comment:
This can be changed to:

```python

if hasattr(self, 'flag'):
    if self.flag.is_flag_enabled:
        return self.flag.state != self.flag.UNFLAGGED
return False

Reply:
Your suggestion is not valid here because the developer may disable flagging and `is_flagged` will still be true and the comments will appear as flagged even if the feature is disabled.

Reply:
I'm not sure I understand your argument since I've not seen all the changes in depth. What I suggested was a simple refactoring of your code, which does the exact same thing as your code or does it?

Reply:
Simply: `self.flag.state != self.flag.UNFLAGGED` can be False or True, even if the flagging system is disabled.
I want `is_flagged` to be always false when the flagging system is disabled.

Reply:
Okay, now I understand. Ihas mistakenly added the `hasattr` check in my comment. I have [edited](https://github.com/Radi85/Comment/pull/35#discussion_r444945408) it now. Everything else is fine.